card/lib/card/query/sql_statement.rb

```diff
@@ -53,27 +55,63 @@ def fields
       end
 
       def joins query
-        [ join_clause(query),
-          query.subqueries.map { |sq| joins sq }
-        ].flatten * "\n"
+        #binding.pry
+        if query.left_joined?
+          join_on_clause query, query.joins.first
+        else
+          [join_on_clause(query, query.joins),
+           query.subqueries.map { |sq| joins sq }
+          ].flatten * "\n"
+        end
+      end
+
+      def join_on_clause query, ready_joins
+        Array.wrap(ready_joins).map do |join|
+          [join_clause(query,join),
+           'ON',
+           on_clause(query, join)
+          ] * ' '
+        end
+      end
+
+      def join_clause query, join
```

Comment:
Assignment Branch Condition size for join_clause is too high. [20.83/15]<br>Use def with parentheses when there are parameters.